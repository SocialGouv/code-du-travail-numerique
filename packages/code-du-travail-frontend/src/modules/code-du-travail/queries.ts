import { SOURCES } from "@socialgouv/cdtn-utils";
import { ElasticLaborCodeArticle } from "./type";
import { DocumentElasticResult, fetchDocument } from "../documents";

export const fetchLegalArticle = async (
  slug: string
): Promise<DocumentElasticResult<ElasticLaborCodeArticle> | undefined> => {
  return await fetchDocument<
    ElasticLaborCodeArticle,
    keyof DocumentElasticResult<ElasticLaborCodeArticle>
  >(["description", "title", "dateDebut", "html", "url", "notaHtml"], {
    query: {
      bool: {
        filter: [
          { term: { source: SOURCES.CDT } },
          { term: { slug } },
          { term: { isPublished: true } },
        ],
      },
    }
  });
};
