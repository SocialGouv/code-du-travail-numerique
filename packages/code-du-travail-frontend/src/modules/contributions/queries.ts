import * as Sentry from "@sentry/nextjs";
import { elasticDocumentsIndex, elasticsearchClient } from "../../api/utils";
import { SOURCES } from "@socialgouv/cdtn-utils";
import { DocumentElasticResult, fetchDocument, isSource } from "../documents";
import { Contribution, ContributionElasticDocument } from "./type";

export const fetchContributions = async <
  K extends keyof ContributionElasticDocument,
>(
  fields: K[],
  filters?: {
    cdtnIds?: string[];
  }
): Promise<Pick<ContributionElasticDocument, K>[]> => {
  const baseFilters: Array<any> = [
    { term: { source: SOURCES.CONTRIBUTIONS } },
    { term: { isPublished: true } },
  ];

  if (filters?.cdtnIds) {
    baseFilters.push({ terms: { cdtnId: filters.cdtnIds } });
  }

  const result = await elasticsearchClient.search<
    Pick<ContributionElasticDocument, K>
  >({
    query: {
      bool: {
        filter: baseFilters,
      },
    },
    size: 3000,
    _source: fields,
    index: elasticDocumentsIndex,
  });

  return result.hits.hits
    .map(({ _source }) => _source)
    .filter((source) => source !== undefined);
};

const formatContribution = (
  contribution: ContributionElasticDocument | undefined
): Contribution | undefined => {
  if (!contribution) {
    return undefined;
  }
  return {
    ...contribution,
    isGeneric: contribution.idcc === "0000",
    isNoCDT: contribution?.type === "generic-no-cdt",
    relatedItems: [
      {
        title: "Articles liÃ©s",
        items: contribution.linkedContent.reduce((arr, linked) => {
          if (!isSource(linked.source)) {
            Sentry.captureMessage(
              `la source de ${JSON.stringify(linked)} est incorrecte`
            );
            return arr;
          }
          return [
            ...arr,
            {
              title: linked.title,
              url: linked.slug,
              source: linked.source,
            },
          ];
        }, []),
      },
    ],
  };
};

export const fetchContributionBySlug = async (
  slug: string
): Promise<Contribution | undefined> => {
  const response = await fetchDocument<
    ContributionElasticDocument,
    keyof DocumentElasticResult<ContributionElasticDocument>
  >(
    [
      "metas",
      "idcc",
      "date",
      "title",
      "slug",
      "type",
      "linkedContent",
      "breadcrumbs",
      "ccSupported",
      "ccUnextended",
      "messageBlock",
      "references",
      "ccnShortTitle",
      "raw",
      "ficheSpDescription",
      "content",
      "url",
      "messageBlockGenericNoCDT",
    ],
    {
      index: elasticDocumentsIndex,
      query: {
        bool: {
          filter: [
            { term: { source: SOURCES.CONTRIBUTIONS } },
            { term: { slug } },
            { term: { isPublished: true } },
          ],
        },
      },
      size: 1,
    }
  );
  return formatContribution(response);
};
