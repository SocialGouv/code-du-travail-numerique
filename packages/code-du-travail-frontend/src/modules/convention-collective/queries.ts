import { ElasticAgreement } from "@socialgouv/cdtn-types";
import { orderByAlpha } from "../utils";
import { SOURCES } from "@socialgouv/cdtn-utils";
import { elasticDocumentsIndex, elasticsearchClient } from "../../api/utils";

type Props<K> = {
  fields?: K[];
  sortBy?: K;
  filterTitle?: string;
  size?: number;
};

export const fetchAllAgreements = async <K extends keyof ElasticAgreement>({
  fields,
  sortBy,
  filterTitle,
  size = 100,
}: Props<K>): Promise<Pick<ElasticAgreement, K>[]> => {
  const response = await elasticsearchClient.search<Pick<ElasticAgreement, K>>({
    query: {
      bool: {
        ...(filterTitle
          ? {
              should: [
                {
                  match: {
                    "shortTitle.french": {
                      boost: 0.9,
                      fuzziness: "1",
                      query: filterTitle,
                    },
                  },
                },
                {
                  match_phrase_prefix: {
                    "synonymes.french": {
                      query: filterTitle,
                    },
                  },
                },
                {
                  match_phrase_prefix: {
                    "title.french": {
                      query: filterTitle,
                    },
                  },
                },
              ],
            }
          : {}),
        filter: [
          { term: { source: SOURCES.CCN } },
          { term: { isPublished: true } },
          { term: { contributions: true } },
        ],
      },
    },
    size,
    ...(fields ? { _source: fields } : {}),
    index: elasticDocumentsIndex,
  });

  const data = response.hits.hits
    .map(({ _source }) => _source)
    .filter((item) => item !== undefined);
  if (sortBy) {
    data.sort((a, b) => orderByAlpha(a, b, sortBy));
  }
  return data;
};
