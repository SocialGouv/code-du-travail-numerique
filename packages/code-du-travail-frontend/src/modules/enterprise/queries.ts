import debounce from "debounce-promise";
import { SITE_URL } from "../../config";
import { Enterprise } from "./types";

export type SearchParams = {
  query: string;
  codesPostaux?: string[];
};

export const messageFetchSearchCcOrEnterprise =
  "Ce service est momentanément indisponible. Si vous utilisez un simulateur, vous pouvez tout de même poursuivre la simulation pour obtenir le résultat prévu par le Code du travail en sélectionnant l'option \"Je ne souhaite pas renseigner ma convention collective (je passe l'étape).\"";

const siretSirenError =
  "Veuillez indiquer un numéro Siret (14 chiffres) ou Siren (9 chiffres) valide";

const siretLengthError =
  "Veuillez indiquer un numéro Siret (14 chiffres obligatoire)";

const siretNumberError =
  "Veuillez indiquer un numéro Siret (14 chiffres uniquement)";

const apiEnterprises = function createFetcher(
  searchParams: SearchParams
): Promise<Enterprise[]> {
  if (
    /^\d{2,8}$/.test(searchParams.query.replace(/\s/g, "")) ||
    /^\d{4}[A-Za-z]$/.test(searchParams.query.replace(/\W/g, ""))
  ) {
    return Promise.reject(siretSirenError);
  }
  if (
    /^\d{10,13}$/.test(searchParams.query.replace(/\s/g, "")) ||
    /^\d{15,}$/.test(searchParams.query.replace(/\s/g, ""))
  ) {
    return Promise.reject(siretLengthError);
  }
  if (/\D+\d{14}/.test(searchParams.query.replace(/\s/g, ""))) {
    return Promise.reject(siretNumberError);
  }

  const url = `http://localhost:3000/api/enterprises?q=${encodeURIComponent(searchParams.query)}${
    searchParams.codesPostaux
      ? `&cp=${encodeURIComponent(searchParams.codesPostaux.join(","))}`
      : ""
  }`;

  return fetch(url)
    .then(async (response) => {
      if (response.ok) {
        const res = await response.json();
        return res.entreprises;
      }
      throw new Error();
    })
    .catch(() => {
      return Promise.reject(messageFetchSearchCcOrEnterprise);
    });
};

const searchEnterprises = debounce(apiEnterprises, 300);

export { searchEnterprises };
