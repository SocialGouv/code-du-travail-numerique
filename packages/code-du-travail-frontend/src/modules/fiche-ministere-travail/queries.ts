import { SOURCES } from "@socialgouv/cdtn-utils";

import { DocumentElasticResult, fetchDocument } from "../documents";
import {
  ElasticFicheTravailEmploi,
  ElasticFicheTravailEmploiSection,
} from "@socialgouv/cdtn-types";

export type FicheMT = DocumentElasticResult<ElasticFicheTravailEmploi> & {
  highlight?: ElasticFicheTravailEmploiSection;
  sections: ElasticFicheTravailEmploiSection[];
};

const formatFiche = (
  fiche: DocumentElasticResult<ElasticFicheTravailEmploi> | undefined
): FicheMT | undefined => {
  if (!fiche) {
    return undefined;
  }
  return {
    ...fiche,
    highlight: fiche.sections.find(
      (section) => !section.anchor || section.anchor === ""
    ),
    sections: fiche.sections.filter(
      (section) => section.anchor && section.anchor !== ""
    ),
  };
};

export const fetchFicheMT = async (
  slug: string
): Promise<FicheMT | undefined> => {
  return formatFiche(
    await fetchDocument<
      ElasticFicheTravailEmploi,
      keyof DocumentElasticResult<ElasticFicheTravailEmploi>
    >(
      [
        "breadcrumbs",
        "date",
        "description",
        "intro",
        "sections",
        "slug",
        "title",
        "url",
        "cdtnId",
      ],
      {
        query: {
          bool: {
            filter: [
              { term: { source: SOURCES.SHEET_MT_PAGE } },
              { term: { slug } },
              { term: { isPublished: true } },
            ],
          },
        },
        size: 1,
      }
    )
  );
};
