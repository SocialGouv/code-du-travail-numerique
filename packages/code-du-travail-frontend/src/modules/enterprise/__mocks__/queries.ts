import { SearchParams } from "../queries";
import { Enterprise } from "../types";

export const messageFetchSearchCcOrEnterprise =
  "Ce service est momentanément indisponible. Si vous utilisez un simulateur, vous pouvez tout de même poursuivre la simulation pour obtenir le résultat prévu par le Code du travail en sélectionnant l'option \"Je ne souhaite pas renseigner ma convention collective (je passe l'étape).\"";

export const searchEnterprises = (
  searchParams: SearchParams
): Promise<Enterprise[]> => {
  const query = searchParams.query.toLowerCase();

  if (query.includes("carrefour")) {
    return Promise.resolve([
      {
        activitePrincipale: "47.11F - Hypermarchés",
        etablissements: 208,
        highlightLabel: "CARREFOUR HYPERMARCHES",
        label: "CARREFOUR HYPERMARCHES",
        simpleLabel: "CARREFOUR HYPERMARCHES",
        matching: 208,
        siren: "451321335",
        address:
          "ZAE SAINT GUENAULT 1 RUE JEAN MERMOZ 91000 EVRY-COURCOURONNES",
        firstMatchingEtablissement: {
          siret: "45132133500023",
          address:
            "ZAE SAINT GUENAULT 1 RUE JEAN MERMOZ 91000 EVRY-COURCOURONNES",
        },
        complements: {
          liste_idcc: ["2216", "1486"],
        },
        conventions: [
          {
            id: "2216",
            contributions: true,
            num: 2216,
            shortTitle:
              "Commerce de détail et de gros à prédominance alimentaire",
            title:
              "Convention collective nationale du commerce de détail et de gros à prédominance alimentaire du 12 juillet 2001.  Etendue par arrêté du 26 juillet 2002 JORF 6 août 2002.",
            url: "https://www.legifrance.gouv.fr/affichIDCC.do?idConvention=KALICONT000005635085",
            slug: "2216-commerce-de-detail-et-de-gros-a-predominance-alimentaire",
          },
          {
            id: "1486",
            contributions: true,
            num: 1486,
            shortTitle:
              "Bureaux d'études techniques, cabinets d'ingénieurs-conseils et sociétés de conseils",
            title:
              "Bureaux d'études techniques, cabinets d'ingénieurs-conseils et sociétés de conseils",
            url: "https://www.legifrance.gouv.fr/affichIDCC.do?idConvention=KALICONT000005635173",
            slug: "1486-bureaux-detudes-techniques-cabinets-dingenieurs-conseils-et-societes-de",
          },
        ],
      },
    ]);
  }

  if (query.includes("bricomanie")) {
    return Promise.resolve([
      {
        activitePrincipale:
          "47.52Z - Commerce de détail de quincaillerie, peintures et verres en magasins spécialisés",
        etablissements: 85,
        highlightLabel: "BRICOMANIE",
        label: "BRICOMANIE",
        simpleLabel: "BRICOMANIE",
        matching: 85,
        siren: "123456789",
        address: "123 AVENUE DU BRICOLAGE 75000 PARIS",
        firstMatchingEtablissement: {
          siret: "12345678900123",
          address: "123 AVENUE DU BRICOLAGE 75000 PARIS",
        },
        complements: {
          liste_idcc: [],
        },
        conventions: [],
      },
    ]);
  }

  return Promise.resolve([]);
};
