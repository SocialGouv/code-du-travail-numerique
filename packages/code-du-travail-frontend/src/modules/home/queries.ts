import { getLabelBySource, getRouteBySource } from "@socialgouv/cdtn-utils";
import { fetchRootThemes } from "../themes";
import { fetchHighLights } from "../highlights";
import { fetchTools } from "../outils";
import { fetchModels } from "../modeles-de-courriers";
import { fetchContributions } from "../contributions";
import { fetchAgreements } from "../convention-collective";

export type HomeCardItem = {
  description: string;
  link: string;
  theme: string;
  title: string;
};

export type HomeTileItem = {
  iconName: string;
  link: string;
  title: string;
  description?: string;
};

export type HomePageProps = {
  highlights: HomeCardItem[];
  tools: HomeTileItem[];
  modeles: HomeCardItem[];
  contributions: HomeCardItem[];
  agreements: HomeCardItem[];
  themes: HomeTileItem[];
};

export const fetchHomeData = async (): Promise<HomePageProps> => {
  const highlights = await fetchHighLights();
  const parsedHighlights = highlights.map((highlight) => ({
    description: highlight.description,
    link: `/${getRouteBySource(highlight.source)}/${highlight.slug}`,
    theme: getLabelBySource(highlight.source),
    title: highlight.title,
  }));

  const tools = await fetchTools(
    ["icon", "source", "slug", "title", "description"],
    {
      cdtnIds: [
        "d7ad36850a", // simulateur-embauche
        "d8a3605790", // indemnitÃ©-licenciement
        "1eea193273", // preavis-demission
        "db8ffe3574", // convention-collective
      ],
    }
  );
  const parsedTools = tools.map((tool) => ({
    iconName: tool.icon,
    link: `/${getRouteBySource(tool.source)}/${tool.slug}`,
    title: tool.title,
    description: tool.description,
  }));

  const modeles = await fetchModels(
    ["source", "slug", "title", "description"],
    {
      cdtnIds: [
        "72bd2d0080", // rupture-du-contrat-en-periode-dessai-par-le-salarie
        "772cb955ce", // rupture-de-periode-dessai-par-lemployeur
        "8122c6c3eb", // convocation-a-un-entretien-prealable-au-licenciement-pour-motif-personnel
        "9a6cf1b40c", // lettre-de-demission
      ],
    }
  );
  const parsedModeles = modeles.map((modele) => ({
    description: modele.description,
    link: `/${getRouteBySource(modele.source)}/${modele.slug}`,
    theme: getLabelBySource(modele.source),
    title: modele.title,
  }));

  const contributions = await fetchContributions(
    ["source", "slug", "title", "description"],
    {
      cdtnIds: [
        "90c0113ee8", // en-cas-darret-maladie-du-salarie-lemployeur-doit-il-assurer-le-maintien-de-salaire
        "db4f1fe3fb", // quelle-est-la-duree-du-preavis-en-cas-de-demission
        "eba7a4592f", // les-conges-pour-evenements-familiaux
        "f6247840b2", // est-il-obligatoire-davoir-un-contrat-de-travail-ecrit-et-signe
      ],
    }
  );
  const parsedContributions = contributions.map((contribution) => ({
    description: contribution.description,
    link: `/${getRouteBySource(contribution.source)}/${contribution.slug}`,
    theme: getLabelBySource(contribution.source),
    title: contribution.title,
  }));

  const agreements = await fetchAgreements(
    ["source", "slug", "shortTitle", "description"],
    undefined,
    {
      cdtnIds: [
        "39ac98db5d", // 573-commerces-de-gros
        "81c96604dc", // 2609-batiment-etam
        "2f57b6af7c", // 3248-metallurgie
        "d825ef1df2", // 3239-particuliers-employeurs-et-emploi-a-domicile
      ],
    }
  );
  const parsedAgreements = agreements.map((agreement) => ({
    description: agreement.description,
    link: `/${getRouteBySource(agreement.source)}/${agreement.slug}`,
    theme: getLabelBySource(agreement.source),
    title: agreement.shortTitle,
  }));

  const themes = await fetchRootThemes(["title", "icon", "slug", "source"]);
  const parsedThemes = themes.map((theme) => ({
    iconName: theme.icon!,
    link: `/${getRouteBySource(theme.source)}/${theme.slug}`,
    title: theme.title,
  }));

  const response = {
    themes: parsedThemes,
    highlights: parsedHighlights,
    tools: parsedTools,
    modeles: parsedModeles,
    contributions: parsedContributions,
    agreements: parsedAgreements,
  };

  return response;
};
