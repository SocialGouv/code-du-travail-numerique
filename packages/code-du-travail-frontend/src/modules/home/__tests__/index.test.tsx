import { render } from "@testing-library/react";
import React from "react";
import { Home } from "..";
import { HomePageProps } from "../queries";

const mockData: HomePageProps = {
  highlights: [
    {
      description: "Highlight description",
      link: "https://example.com/highlight",
      theme: "light",
      title: "Highlight Title",
    },
  ],
  tools: [
    {
      iconName: "tool-icon",
      link: "https://example.com/tool",
      title: "Tool Title",
      description: "Tool description",
    },
  ],
  modeles: [
    {
      description: "Model description",
      link: "https://example.com/model",
      theme: "dark",
      title: "Model Title",
    },
  ],
  contributions: [
    {
      description: "Contribution description",
      link: "https://example.com/contribution",
      theme: "light",
      title: "Contribution Title",
    },
  ],
  agreements: [
    {
      description: "Agreement description",
      link: "https://example.com/agreement",
      theme: "dark",
      title: "Agreement Title",
    },
  ],
  themes: [
    {
      iconName: "theme-icon",
      link: "https://example.com/theme",
      title: "Theme Title",
      description: "Theme description",
    },
  ],
};

describe("<Home />", () => {
  it("should match snapshot", () => {
    const { container } = render(<Home {...mockData} />);
    expect(container).toMatchSnapshot();
  });
});
