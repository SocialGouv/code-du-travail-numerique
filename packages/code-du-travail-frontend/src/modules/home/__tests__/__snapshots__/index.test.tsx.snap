// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshot 1`] = `
<div>
  <div
    class="fr-grid-row"
  >
    <div
      class="fr-col-12"
    >
      <div
        class="bg_var(--blue-cumulus-925-125)"
      >
        <div
          class="fr-container fr-py-6w"
        >
          <div
            class="fr-grid-row fr-grid-row--gutters fr-grid-row--middle"
          >
            <div
              class="fr-col-12 fr-col-md-7"
            >
              <h1
                class="fr-mb-2w fr-text--lead fr-text--bold"
              >
                Bienvenue sur
                 
                <span
                  class="fr-mt-2w fr-mb-0 fr-h1 d_block"
                >
                  le Code du travail numérique
                </span>
              </h1>
              <h2
                class="fr-text--lead fr-text--bold fr-mb-6w"
              >
                Obtenez les réponses à vos questions sur le droit du travail.
              </h2>
              <form
                class="fr-grid-row fr-grid-row--gutters fr-grid-row--bottom"
                role="search"
              >
                <div
                  class="fr-col-12 fr-col-md-8"
                >
                  <div
                    class="d_flex flex-d_row ai_center w_100% pos_relative"
                  >
                    <div
                      class="fr-input-group w_100% fr-mb-0"
                    >
                      <label
                        class="fr-label"
                        for="downshift-0-input"
                        id="downshift-0-label"
                      >
                        Recherchez par mots-clés
                        <span
                          class="fr-sr-only"
                        >
                          , la sélection d'une option charge une nouvelle page
                        </span>
                        <span
                          class="fr-hint-text"
                        >
                          par exemple : congés payés, durée de préavis
                        </span>
                      </label>
                      <div
                        class="fr-input-wrap fr-input-wrap--addon"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-expanded="false"
                          autocomplete="off"
                          class="fr-input"
                          data-testid="search-input"
                          id="downshift-0-input"
                          role="combobox"
                          type="text"
                          value=""
                        />
                        <div
                          class="pos_absolute right_0 h_100% ac_center"
                        />
                      </div>
                      <div
                        aria-live="polite"
                        class="fr-messages-group"
                        id="downshift-0-input-messages-group"
                      />
                    </div>
                    <ul
                      aria-labelledby="downshift-0-label"
                      class="fr-p-0 fr-m-0 pos_absolute top_100% left_0 w_100% z_1000 bg_var(--background-default-grey) li-t_none! bd-t_none bx-sh_0_4px_8px_rgba(0,_0,_0,_0.1) max-h_300px ov-y_auto "
                      id="downshift-0-menu"
                      role="listbox"
                    />
                  </div>
                </div>
                <div
                  class="fr-col-12 fr-col-md-4"
                >
                  <button
                    aria-label="Lancer la recherche"
                    class="fr-btn fr-icon-search-line fr-btn--icon-right mdDown:w_100%! mdDown:jc_center!"
                    id="fr-button-«r1»"
                    type="submit"
                  >
                    Rechercher
                  </button>
                </div>
              </form>
            </div>
            <div
              class="fr-col-md-5 fr-col-12"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="331"
                sizes="(max-width: 768px) 378px, 486px"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 16w, /_next/image?url=%2Fimg.jpg&w=32&q=75 32w, /_next/image?url=%2Fimg.jpg&w=48&q=75 48w, /_next/image?url=%2Fimg.jpg&w=64&q=75 64w, /_next/image?url=%2Fimg.jpg&w=96&q=75 96w, /_next/image?url=%2Fimg.jpg&w=128&q=75 128w, /_next/image?url=%2Fimg.jpg&w=256&q=75 256w, /_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="color: transparent; width: 100%; height: auto;"
                width="486"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        id="home-highlights"
      >
        <div
          class="fr-my-8w fr-container"
        >
          <h2
            class="fr-mb-3v"
          >
            À la une
          </h2>
          <div
            class="fr-grid-row fr-grid-row--gutters fr-grid-row--right"
          >
            <div
              class="fr-col-12 fr-col-md-6 fr-col-lg-6"
            >
              <div
                class="fr-card fr-enlarge-link fr-card--horizontal"
                id="fr-card-«r2»"
              >
                <div
                  class="fr-card__body"
                >
                  <div
                    class="fr-card__content"
                  >
                    <h3
                      class="fr-card__title"
                    >
                      <a
                        class=""
                        href="https://example.com/highlight"
                      >
                        Highlight Title
                      </a>
                    </h3>
                    <p
                      class="fr-card__desc"
                    >
                      Highlight description
                    </p>
                    <div
                      class="fr-card__start fr-mb-2w"
                    >
                      <p
                        class="fr-tag fr-tag--sm"
                      >
                        light
                      </p>
                    </div>
                    <div
                      class="fr-card__end fr-text--xs"
                    >
                      <p
                        class="fr-m-0 fr-text--xs"
                      >
                        Consulter
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="bg_var(--background-alt-blue-cumulus)"
        id="home-outils"
      >
        <div
          class="fr-my-8w fr-container fr-py-6w"
        >
          <h2>
            Boîte à outils
          </h2>
          <p
            class="fr-text--xl"
          >
            Trouvez des réponses personnalisées selon votre situation
          </p>
          <div
            class="fr-grid-row fr-grid-row--gutters fr-grid-row--right"
          >
            <div
              class="fr-col-12 fr-col-sm-6 fr-col-md-6 fr-col-lg-3"
            >
              <div
                class="fr-tile fr-enlarge-link fr-tile--vertical"
                id="fr-tile-«r3»"
              >
                <div
                  class="fr-tile__body"
                >
                  <div
                    class="fr-tile__content"
                  >
                    <h3
                      class="fr-tile__title"
                    >
                      <a
                        aria-disabled="false"
                        class=""
                        href="https://example.com/tool"
                      >
                        Tool Title
                      </a>
                    </h3>
                    <div
                      class="fr-tile__desc"
                    >
                      <p>
                        Tool description
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="fr-tile__header"
                >
                  <div
                    class="fr-tile__img fr-mb-0"
                  >
                    <img
                      alt=""
                      class="fr-responsive-img"
                      height="80"
                      src="/static/assets/icons/tools/tool-icon.svg"
                      width="80"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="fr-mt-3w d_flex jc_center"
          >
            <a
              class="fr-btn fr-btn--secondary fr-btn--lg fr-icon-arrow-right-line fr-btn--icon-right"
              href="/outils"
              id="fr-button-«r4»"
            >
              Voir tous les outils
            </a>
          </div>
        </div>
      </div>
      <div
        id="home-modeles-de-courriers"
      >
        <div
          class="fr-my-8w fr-container"
        >
          <h2>
            Modèles de documents
          </h2>
          <p
            class="fr-text--xl"
          >
            Téléchargez et personnalisez les modèles de documents et de lettres pour vos démarches en lien avec le droit du travail
          </p>
          <div
            class="fr-grid-row fr-grid-row--gutters fr-grid-row--right"
          >
            <div
              class="fr-col-12 fr-col-md-6 fr-col-lg-6"
            >
              <div
                class="fr-card fr-enlarge-link fr-card--horizontal"
                id="fr-card-«r5»"
              >
                <div
                  class="fr-card__body"
                >
                  <div
                    class="fr-card__content"
                  >
                    <h3
                      class="fr-card__title"
                    >
                      <a
                        class=""
                        href="https://example.com/model"
                      >
                        Model Title
                      </a>
                    </h3>
                    <p
                      class="fr-card__desc"
                    >
                      Model description
                    </p>
                    <div
                      class="fr-card__start fr-mb-2w"
                    >
                      <p
                        class="fr-tag fr-tag--sm"
                      >
                        dark
                      </p>
                    </div>
                    <div
                      class="fr-card__end fr-text--xs"
                    >
                      <p
                        class="fr-m-0 fr-text--xs"
                      >
                        Consulter
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="fr-mt-3w d_flex jc_center"
          >
            <a
              class="fr-btn fr-btn--secondary fr-btn--lg fr-icon-arrow-right-line fr-btn--icon-right"
              href="/modeles-de-courriers"
              id="fr-button-«r6»"
            >
              Voir tous les modèles de documents
            </a>
          </div>
        </div>
      </div>
      <div
        id="home-fiches-pratiques"
      >
        <div
          class="fr-my-8w fr-container"
        >
          <h2>
            Vos fiches pratiques
          </h2>
          <p
            class="fr-text--xl"
          >
            Obtenez une réponse personnalisée selon votre convention collective
          </p>
          <div
            class="fr-grid-row fr-grid-row--gutters fr-grid-row--right"
          >
            <div
              class="fr-col-12 fr-col-md-6 fr-col-lg-6"
            >
              <div
                class="fr-card fr-enlarge-link fr-card--horizontal"
                id="fr-card-«r7»"
              >
                <div
                  class="fr-card__body"
                >
                  <div
                    class="fr-card__content"
                  >
                    <h3
                      class="fr-card__title"
                    >
                      <a
                        class=""
                        href="https://example.com/contribution"
                      >
                        Contribution Title
                      </a>
                    </h3>
                    <p
                      class="fr-card__desc"
                    >
                      Contribution description
                    </p>
                    <div
                      class="fr-card__start fr-mb-2w"
                    >
                      <p
                        class="fr-tag fr-tag--sm"
                      >
                        light
                      </p>
                    </div>
                    <div
                      class="fr-card__end fr-text--xs"
                    >
                      <p
                        class="fr-m-0 fr-text--xs"
                      >
                        Consulter
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="fr-mt-3w d_flex jc_center"
          >
            <a
              class="fr-btn fr-btn--secondary fr-btn--lg fr-icon-arrow-right-line fr-btn--icon-right"
              href="/contribution"
              id="fr-button-«r8»"
            >
              Voir toutes les fiches pratiques
            </a>
          </div>
        </div>
      </div>
      <div
        class="bg_var(--background-alt-blue-cumulus)"
        id="home-convention-collective"
      >
        <div
          class="fr-my-8w fr-container fr-py-6w"
        >
          <h2>
            Votre convention collective
          </h2>
          <p
            class="fr-text--xl"
          >
            Retrouvez les questions-réponses fréquentes organisées par thème pour votre convention collective
          </p>
          <div
            class="fr-grid-row fr-grid-row--gutters fr-grid-row--right"
          >
            <div
              class="fr-col-12 fr-col-md-6 fr-col-lg-3"
            >
              <div
                class="fr-card fr-enlarge-link fr-card--horizontal"
                id="fr-card-«r9»"
              >
                <div
                  class="fr-card__body"
                >
                  <div
                    class="fr-card__content"
                  >
                    <h3
                      class="fr-card__title"
                    >
                      <a
                        class=""
                        href="https://example.com/agreement"
                      >
                        Agreement Title
                      </a>
                    </h3>
                    <p
                      class="fr-card__desc"
                    >
                      Agreement description
                    </p>
                    <div
                      class="fr-card__start fr-mb-2w"
                    >
                      <p
                        class="fr-tag fr-tag--sm"
                      >
                        dark
                      </p>
                    </div>
                    <div
                      class="fr-card__end fr-text--xs"
                    >
                      <p
                        class="fr-m-0 fr-text--xs"
                      >
                        Consulter
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="fr-mt-3w d_flex jc_center"
          >
            <a
              class="fr-btn fr-btn--secondary fr-btn--lg fr-icon-arrow-right-line fr-btn--icon-right"
              href="/convention-collective"
              id="fr-button-«ra»"
            >
              Voir toutes les conventions collectives
            </a>
          </div>
        </div>
      </div>
      <div
        id="home-themes"
      >
        <div
          class="fr-my-8w fr-container"
        >
          <h2>
            Thèmes
          </h2>
          <p
            class="fr-text--xl"
          >
            Retrouvez tous nos contenus organisés par thèmes
          </p>
          <ul
            class="fr-grid-row fr-grid-row--gutters fr-grid-row--right"
          >
            <li
              class="fr-col-4 fr-hidden fr-unhidden-lg li-s_none!"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="233"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=828&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=828&q=75 2x"
                style="color: transparent;"
                width="384"
              />
            </li>
            <li
              class="fr-col-12 fr-col-sm-6 fr-col-md-4 fr-col-lg-2 li-s_none!"
            >
              <div
                class="fr-tile fr-enlarge-link fr-tile--no-icon fr-tile--sm"
              >
                <p
                  class="fr-tile__title"
                >
                  <a
                    href="https://example.com/theme"
                  >
                    Theme Title
                  </a>
                </p>
                <div
                  class="fr-tile__img"
                >
                  <img
                    alt=""
                    data-nimg="1"
                    decoding="async"
                    height="56"
                    loading="lazy"
                    src="/static/assets/icons/themes/theme-icon.svg"
                    style="color: transparent;"
                    width="56"
                  />
                </div>
              </div>
            </li>
          </ul>
          <div
            class="fr-mt-3w d_flex jc_center"
          >
            <a
              class="fr-btn fr-btn--secondary fr-btn--lg fr-icon-arrow-right-line fr-btn--icon-right"
              href="/themes"
              id="fr-button-«rb»"
            >
              Voir tous les thèmes
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
